File: front/src/components/authentication/dtos/user-auth-profile-dto.ts
import { AddressDto } from "shared/components/address/dtos/address-dto";
import { ContactDto } from "shared/components/contact/dtos/contact-dto";
import { RootBase } from "shared/entities-dtos/root-base";
import { UserAccountAuthDto } from "./user-account-auth-dto";
import { UserProfileDto } from "./user-profile-dto";

export class UserAuthProfileDto extends RootBase {

  // email!: string;
  // displayUserName!: string;
  // userName!: string;
  // address!: AddressDto;
  // contact!: ContactDto;
  userAccountAuth!: UserAccountAuthDto;
  userAccountProfile!: UserProfileDto;


}

File: back/Application/Services/Operations/Auth/Dtos/UserAuthProfileDto.cs
using Application.Services.Operations.Profiles.Dtos;
using Application.Services.Shared.Dtos;

namespace Application.Services.Operations.Auth.Dtos;

// public class UserAccountProfileDto : RootBaseDto
// {
//     public required string Email { get; set; }
//     public required string DisplayUserName { get; set; }
//     public required string UserName { get; set; }
//     public required AddressDto Address { get; set; }
//     public required ContactDto Contact { get; set; }

// }
public class UserAuthProfileDto : RootBaseDto
{
    public override int Id { get; set; } = -1;
    public required UserAccountDto UserAccountAuth { get; set; }
    public required UserProfileDto UserAccountProfile { get; set; }

}